etc/afpd/uam.c

```diff
@@ -210,7 +210,11 @@ struct passwd *uam_getname(void *private, char *name, const int len)
             princ = bformat("%s@%s", name, obj->options.addomain);
         else
             princ = bformat("%s%s%s", obj->options.ntdomain, obj->options.ntseparator, name);
-        pwent = getpwnam(bdata(princ));
+        if (bdata(princ) != NULL) {
```

Comment:
Side note: There might be a case for adding some warning or debug logging somewhere around here. In this branch, the user has clearly tried to register with a Microsoft domain, but the input data is bad.

Anyways, out of scope for this PR.